app/src/main/java/com/cupcake/chickenmasala/ui/fragment/details/DetailsFragment.kt

```diff
@@ -1,56 +1,63 @@
-import com.cupcake.chickenmasala.ui.fragment.details.adapter.InstructionsAdapter
+import com.cupcake.chickenmasala.databinding.FragmentDetailsBinding
+import com.cupcake.chickenmasala.ui.base.BaseFragment
+import com.cupcake.chickenmasala.ui.fragment.details.adapter.DetailsAdapter
+import com.cupcake.chickenmasala.ui.fragment.details.adapter.DetailsItem
+import com.cupcake.chickenmasala.ui.fragment.details.adapter.DetailsItemType
 import com.cupcake.chickenmasala.usecase.Repository
 import com.cupcake.chickenmasala.utill.DataSourceProvider
-import com.cupcake.chickenmasala.utill.setImage
 
 
 class DetailsFragment : BaseFragment<FragmentDetailsBinding>() {
 
     override val LOG_TAG: String = "Details_Fragment"
-    private lateinit var repository: Repository
-    private lateinit var ingredientsAdapter: IngredientsAdapter
-    private lateinit var instructionsAdapter: InstructionsAdapter
-    private lateinit var cleanIngredientsAdapter: CleanIngredientsAdapter
+
 
     override val bindingInflater: (LayoutInflater, ViewGroup, Boolean) -> FragmentDetailsBinding =
         FragmentDetailsBinding::inflate
 
+    private val repository: Repository by lazy {
+        RepositoryImpl(DataSourceProvider.getDataSource(requireActivity().application))
+    }
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        repository = RepositoryImpl(DataSourceProvider.getDataSource(requireActivity().application))
-        val id = arguments.let { it?.getInt(ID) }
-        setupRecipeDetails(getRecipeById(id!!))
 
+        val idRecipe = arguments.let { it?.getInt(ID) }
+        idRecipe?.let { setup(getRecipeById(it)) }
 
     }
 
-    private fun setupRecipeDetails(recipe: Recipe) {
-        setDataToUiViews(recipe)
-        setupIngredientsRecycleView(recipe.translatedIngredients)
-        setupInstructionsRecycleView(recipe.translatedInstructions)
-        setupCleanedIngredientsRecycleView(recipe.cleanedIngredients)
+    private fun setup(recipe: Recipe) {
+        setUpRecyclerDetails(recipe)
         setupBackButton()
         setUpShareButton(recipe.urlDetailsRecipe)
+    }
+
+    private fun setUpRecyclerDetails(recipe: Recipe) {
+
+        val itemList: MutableList<DetailsItem<Any>> = mutableListOf()
 
+        itemList.add(DetailsItem(recipe, DetailsItemType.INFO))
+        itemList.add(DetailsItem(recipe.translatedIngredients, DetailsItemType.STEP_INGREDIENTS))
+        itemList.add(DetailsItem(recipe.translatedInstructions, DetailsItemType.STEP_INSTRUCTIONS))
+
+        val adapter = DetailsAdapter(itemList)
+        binding.recycler.adapter = adapter
     }
 
-    private fun setUpShareButton(url:String) {
+    private fun setUpShareButton(url: String) {
         binding.toolBar.setOnMenuItemClickListener {
-            when(it.itemId){
-                R.id.icon_share ->{
+            when (it.itemId) {
```

Comment:
use if instead it's just one case!

Reply:
I didn't write this code. This is old code.